mne/decoding/time_gen.py

```diff
@@ -625,39 +707,34 @@ def _sliding_window(times, window_params, sfreq):
         List of training slices, indicating for each classifier the time
         sample (in indices of times) to be fitted on.
     """
+    import copy
 
-    window_params = _DecodingTime(window_params)
+    window = _DecodingTime(copy.deepcopy(window))
 
     # Default values
-    if ('slices' in window_params and
-            all(k in window_params for k in
-                ('start', 'stop', 'step', 'length'))):
-        time_pick = window_params['slices']
+    has_slice = 'slices' in window
+    if has_slice:
+        time_pick = window['slices']
```

Comment:
@jona-sassenhagen if you know the code / understand it better than me, feel free to carefully review it.